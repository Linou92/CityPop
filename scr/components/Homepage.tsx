// import React from 'react';
// import {View, Text, StyleSheet, ImageBackground} from 'react-native';
// import Background from '../images/6.png'

// interface Props {
//   title: string;
// }

// const Homepage = () => { // main page UI
//   return (
//     // Whole page UI
//     <View style={styles.container}> 

//       <ImageBackground source={Background} style={styles.background}>

//         <Text style={styles.titleText}>CityPop</Text>
//         <Text style={styles.subTitleText}>Welcome to CityPop ! You can check the population count of the cities all around the world !</Text>

//         <View>
//           <Text style={styles.button}>Search by city button</Text>
//         </View>

//         <View>
//           <Text style={styles.button}>Search by country button</Text>
//         </View>

//       </ImageBackground>

//     </View>
//   );
// };

// const styles = StyleSheet.create({
//   container: {
//     paddingVertical: 20,
//     borderBottomWidth: 1,
//     borderBottomColor: '#cfcfcf',
//     alignItems: 'center',
//   },
//   titleText: {
//     fontSize: 50,
//     color: '#FFF',
//   },
//   subTitleText: {
//     fontSize: 20,
//     color: '#FFF',
//   },
//   button: {
//     width: 150,
//     height: 110,
//     alignItems: 'center',
//   },
//   background: {
//     //width: '100%',
//     height: '100%',
//     alignItems: 'center',
    
//   }
// });

// export default Homepage;