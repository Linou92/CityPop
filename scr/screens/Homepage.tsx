import React from "react";
import { View, Text, StyleSheet, Image, ImageBackground } from "react-native";
import Button from "../components/Buttons"

export const Homepage = ({ navigation }) => {
    return (
        <View style={styles.container}>
            
            <View style={styles.title}>
                <Text style={styles.titleText}> CityP<Image source={require('../images/logo.png')} style={styles.logoIcon}/>p </Text>
            </View>
            
            <View style={styles.subTitle}>
                <Text style={styles.subTitleText}> Welcome to CityPop ! {"\n"} You can check the population count {"\n"} of the cities all around the world ! </Text>
            </View>

            <View style={styles.button}>
                <Button text="SEARCH BY CITY" onPress={() => navigation.navigate('SearchByCity')}/>
            </View>

            <View style={styles.button}>
                <Button text="SEARCH BY COUNTRY" onPress={() => navigation.navigate('SearchByCountry')}/>
            </View>

            <View style={styles.footer}>
                <ImageBackground source={require('../images/background.png')}  resizeMode="stretch" style={styles.background}>
                    <Text style={styles.footerText}>By Lina Abu Hijleh - 2022</Text>
                </ImageBackground>
            </View>
            
            
        </View>
    )
}


const styles = StyleSheet.create({

    container: {
        flex: 1,
        backgroundColor: '#e6f5ef',
    },

    title: {
        flex: 4,
        justifyContent: 'center',  
    },

    titleText: {
        flex: 3,
        textAlign: 'center',
        fontSize: 50,
        marginTop: 50,
        color: '#50aab5',
    },

    logoIcon: {
        width:80,
        height: 70,
    },

    subTitle: {
        flex: 3,
        justifyContent: 'center',
        marginTop: 50,
    },

    subTitleText: {
        flex: 3,
        textAlign: 'center',
        color: '#b4ced1',
        fontSize: 17,
    },

    button: {
        flex: 3,
        justifyContent: 'center',
        alignItems: 'center',
    },

    footer: {
        flex: 5,
    },
    
    background: {
        flex: 5,
        justifyContent: "center",
    },

    footerText: {
        fontSize: 13,
        textAlign: "center",
        marginTop: 150,
        color: '#50aab5',
        fontWeight: 'bold'
    }

})