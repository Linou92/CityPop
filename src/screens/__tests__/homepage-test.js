import React from 'react';
import renderer from 'react-test-renderer';
import {Homepage} from '../Homepage';

const mockedDispatch = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
  };
});


describe("HomePage", () => {
  beforeEach(() => {
    mockedDispatch.mockClear();
  });
  test('renders homepage correctly', () => {
    const tree = renderer.create(<Homepage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
