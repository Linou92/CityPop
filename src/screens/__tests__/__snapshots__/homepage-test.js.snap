// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e6f5ef",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 4,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#50aab5",
          "flex": 3,
          "fontSize": 50,
          "marginTop": 50,
          "textAlign": "center",
        }
      }
    >
       CityP
      <Image
        source={
          Object {
            "testUri": "../../../src/images/logo.png",
          }
        }
        style={
          Object {
            "height": 70,
            "width": 80,
          }
        }
      />
      p 
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 3,
        "justifyContent": "center",
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#b4ced1",
          "flex": 3,
          "fontSize": 17,
          "textAlign": "center",
        }
      }
    >
       Welcome to CityPop ! 
      

       You can check the population count 
      

       of the cities all around the world ! 
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#50aab5",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "width": "60%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
           
          SEARCH BY CITY
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#50aab5",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "width": "60%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
           
          SEARCH BY COUNTRY
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 5,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 5,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="stretch"
        source={
          Object {
            "testUri": "../../../src/images/background.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          Object {
            "color": "#50aab5",
            "fontSize": 13,
            "fontWeight": "bold",
            "marginTop": 150,
            "textAlign": "center",
          }
        }
      >
        By Lina Abu Hijleh - 2022
      </Text>
    </View>
  </View>
</View>
`;
